### ISSUE TITLE
Grafana (with protocol = socket) dies after a few seconds

### ISSUE THREAD

**Issue by mario-kr** [NixOS/nixpkgs#207824]
### Describe the bug

`grafana.service` gets killed by systemd after ~1-5 seconds.

TL;DR:
grafana seems to call the syscall fchownat, which is not included in the `SystemCallFilter` in the service-definition, and is therefore getting killed by systemd.

### Steps to reproduce

I had this error originally when I upgraded my homeserver from channel 22.05 to 22.11. However, I was able to reproduce the issue in a freshly-installed VM with a minimal `configuration.nix`:

```
{ config, pkgs, ... }:

{
  imports =
    [ 
      ./hardware-configuration.nix
    ];

  boot.loader.grub.enable = true;
  boot.loader.grub.version = 2;
  boot.loader.grub.device = "/dev/sda";

  networking.hostName = "nixos";
  networking.networkmanager.enable = true;
  time.timeZone = "Europe/Berlin";

  i18n.defaultLocale = "en_US.UTF-8";

  environment.systemPackages = with pkgs; [
    vim
  ];

  services.openssh.enable = true;
  services.openssh.permitRootLogin = "yes";

  services.grafana = {
    enable = true;
    settings.server.protocol = "socket";
  };

  networking.firewall.enable = false;

  system.stateVersion = "22.11";
}
```

or, the relevant part only:
```
  services.grafana = {
    enable = true;
    settings.server.protocol = "socket";
  };
```

### Expected behavior

grafana not getting killed by systemd.

#### `systemctl status grafana.server` (after it died)

```
× grafana.service - Grafana Service Daemon
     Loaded: loaded (/etc/systemd/system/grafana.service; enabled; preset: enabled)
     Active: failed (Result: core-dump) since Mon 2022-12-26 12:46:49 CET; 9min ago
   Duration: 4.805s
    Process: 3446 ExecStartPre=/nix/store/3pqsj937vhbs07gssvzqsffcz73fv0ln-unit-script-grafana-pre-start/bin/grafana-pre-start (code=exited, status=0/SUCCESS)
    Process: 3449 ExecStart=/nix/store/prpxqw0r8yn9s13jqzam6591dk9gqpma-unit-script-grafana-start/bin/grafana-start (code=dumped, signal=SYS)
   Main PID: 3449 (code=dumped, signal=SYS)
         IP: 0B in, 0B out
        CPU: 1.266s

Dec 26 12:46:48 nixos grafana-start[3449]: logger=query_data t=2022-12-26T12:46:48.723547949+01:00 level=info msg="Query Service initialization"
Dec 26 12:46:48 nixos grafana-start[3449]: logger=live.push_http t=2022-12-26T12:46:48.7267476+01:00 level=info msg="Live Push Gateway initialization"
Dec 26 12:46:48 nixos grafana-start[3449]: logger=infra.usagestats.collector t=2022-12-26T12:46:48.944550766+01:00 level=info msg="registering usage stat providers" usageStatsProvidersLen=2
Dec 26 12:46:48 nixos grafana-start[3449]: logger=provisioning.plugins t=2022-12-26T12:46:48.949549282+01:00 level=error msg="Failed to read plugin provisioning files from directory" path=/nix/store/bb3gakvk1hwa8r35r92zbl8kwsswh8nw-grafana-provisioning/plugins error="open /nix/store/bb3gakvk1hwa8r35r92zbl8kwsswh8nw-grafana-provisioning/plugins: no such file or directory"
Dec 26 12:46:48 nixos grafana-start[3449]: logger=provisioning.alerting t=2022-12-26T12:46:48.949925113+01:00 level=info msg="starting to provision alerting"
Dec 26 12:46:48 nixos grafana-start[3449]: logger=provisioning.alerting t=2022-12-26T12:46:48.950004867+01:00 level=info msg="finished to provision alerting"
Dec 26 12:46:48 nixos grafana-start[3449]: logger=grafanaStorageLogger t=2022-12-26T12:46:48.951870111+01:00 level=info msg="storage starting"
Dec 26 12:46:49 nixos systemd[1]: grafana.service: Main process exited, code=dumped, status=31/SYS
Dec 26 12:46:49 nixos systemd[1]: grafana.service: Failed with result 'core-dump'.
Dec 26 12:46:49 nixos systemd[1]: grafana.service: Consumed 1.266s CPU time, no IP traffic.
```

Apparently, `status=31/SYS` means the process got killed, because it tried to use a system call which is not in the allow-list in the service definition.

#### dmesg entry

```
[Mon Dec 26 12:46:48 2022] audit: type=1326 audit(1672055208.951:87): auid=4294967295 uid=196 gid=994 ses=4294967295 subj=kernel pid=3449 comm="grafana-server" exe="/nix/store/9x612gpx42ynlxkf3h88ia2wkk6hr96n-grafana-9.3.1/bin/grafana-server" sig=31 arch=c000003e syscall=260 compat=0 ip=0x40c14e code=0x80000000
```

`syscall=260` seems to be `fchownat`, which indeed is missing in `SystemCallFilter` (via `systemctl show grafana.service`)

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.83, NixOS, 22.11 (Raccoon), 22.11.1069.cbe419ed4c8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```
```
[root@nixos:~]# nix-channel --list
nixos https://nixos.org/channels/nixos-22.11
```

**Comment by mario-kr** [comment]
Seems fixed in #205865

Need to wait for #206899 and then do an upgrade.

To my defense, there were no issues on this topic yet, so the issue-search returned nothing.
